src/storage/src/table/mod.rs

```diff
@@ -23,4 +23,6 @@ use crate::error::StorageResult;
 #[async_trait::async_trait]
 pub trait TableIter: Send {
     async fn next(&mut self) -> StorageResult<Option<Row>>;
+
+    async fn next_with_pk(&mut self) -> StorageResult<Option<(Vec<u8>, Row)>>;
```

Comment:
I think we should only keep the latter one. The implementation of these two methods is almost the same except for the first one ignoring the pk at last.

Reply:
Seems not necessary. They're with the definitely same implementation. 🤪